test/unit/test_threaded_downloader.py

```diff
@@ -312,41 +312,12 @@ def connection_error(*args, **kwargs):
 
             self.assertEqual(report2.state, report2.DOWNLOAD_FAILED)
 
-            expected_log_message = "Connection Error - http://fakeurl/primary.xml " \
-                                   "could not be reached."
+            expected_log_message = "Skipping requests to fakeurl due to repeated " \
+                                   "connection failures: "
             log_calls = [mock_call[1][0] for mock_call in mock_logger.mock_calls]
 
             self.assertIn(expected_log_message, log_calls)
 
-    def test_fetch_with_connection_error_badstatusline(self):
```

Comment:
I decided to remove it as it is no longer something we deal with at all. `urllib3` is responsible for noticing the problem and retrying. It felt incorrect to unit test `urllib3`'s functionality here and there isn't really a neat way to do it, either.